hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -296,12 +297,22 @@ checkClose ctx headContext parties initialUtxoHash snapshotNumber closedUtxoHash
     | otherwise = traceError "negative snapshot number"
 {-# INLINEABLE checkClose #-}
 
--- | Checks that the datum
+-- | Checks that the tx contains lower/upper validity and that their difference
+-- is within reasonable bounds
 makeContestationDeadline :: ContestationPeriod -> ScriptContext -> POSIXTime
 makeContestationDeadline cperiod ScriptContext{scriptContextTxInfo} =
-  case ivTo (txInfoValidRange scriptContextTxInfo) of
-    UpperBound (Finite time) _ -> addContestationPeriod time cperiod
-    _ -> traceError "no upper bound validaty interval defined for close"
+  case (txValidFrom, txValidTo) of
+    (LowerBound (Finite startTime) _, UpperBound (Finite endTime) _) ->
+      -- calculate new upper bound by adding the upper bound to the contestation period
+      let newUpperBound = addContestationPeriod endTime cperiod
+          withinBounds = newUpperBound - startTime < acceptableDifference
```

Comment:
🤓 .

Reply:
This is not what we want to ensure! The goal of this that (as you say above in the haddock) **the upper and lower bound validity is within reasonable bounds**.

So it should be `withinBounds = endtime - startTime < acceptableDifference`

Reply:
yes I think I lost track of what I want to do in the process of doing it! contestation deadline can be whatever we set in the datum but the tx upper vs lower bound is what needs to make sense here.